### PR TITLE
Congrats: Update Entrepreneur/Ecommerce Plan Congrats Page Content

### PR DESCRIPTION
**Pull request by chad1008**
Related to 4188-gh-Automattic/dotcom-forge, https://github.com/Automattic/wp-calypso/pull/79525

## Proposed Changes

- Update the Plans-Only congrats page to show specific header text for the Entrepreneur Plan based on [provided mock up](https://github.com/Automattic/wp-calypso/pull/79525#issuecomment-1663995023)
- Within that text, include a separate flow for accounts with unverified email addresses (see note on this below)
- Remove the `Let's work on the site` CTA button for Entrepreneur Plan purchases

**Before:**
entrepreneur plan 'before' image

"After" images are below in the testing instructions.

## Testing Instructions

**With email verified:**
- Use an account with a verified email address
- Purchase an Entrepreneur Plan
- On the Congrats page, confirm that the updated text appears:
entrepreneur plan 'before' image with a verified email address

- Clicking on the CTA button loads the site's WooCommerce setup page

**With email unverified (please see notes below on re: testing this more thoroughly):**

- Use an account with an unverified email address, or
<details>
<summary>Or apply this super-simple testing diff:</summary>

```diff
diff --git a/client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx b/client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
index 09bf783938..dc6313f657 100644
--- a/client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
+++ b/client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
@@ -86,7 +86,7 @@ export default function PlanOnlyThankYou( {
 	let subtitle;
 	let headerButtons;
 	if ( primaryPurchase.productSlug === 'ecommerce-bundle' ) {
-		if ( isEmailVerified ) {
+		if ( ! isEmailVerified ) {
 			subtitle = translate( "With the plan sorted, it's time to start setting up your store." );
 			headerButtons = (
 				<Button

```

</details>

- Purchase an Entrepreneur plan
- On the Congrats page, confirm that the updated text appears:
entrepreneur plan 'before' image with an unverified email address

- Confirm that the correct email address for the account being used is displayed
- Confirm that clicking on the CTA button should resends a verification email (again, see note below regarding this part)

**For both flows:**
- Confirm that there is no **Let's work on the site** button in the page content. There should only be one button, labeled **Manage Plan**. No changes were made to that button on this PR so don't worry about testing it. Unless you really want, to I suppose. You do you.

## Notes
- There's currently one inline note on something I'd love advice/input on
- It doesn't look like we previously had Tracks events set for these CTAs, (unless i'm just overlooking them) do we want to set some up?
- I had a hard time effectively testing the unverified email flow. I can (as shown in the testing diff above) force the flow to trigger by inverting the `isEmailVerified` check, but in actual testing any account shows in Calypso as verified... even brand new testing accounts that I hadn't even opened the verification email for. I'm not sure if the account is being auto-verified somehow (I feel like I remember that being a thing a few years back but can't find a trace of it on MGS) or if I'm just not detecting the unverified state properly. If anyone has tips on how to more accurately test the unverified email flow I'd be super grateful. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
Comment (chad1008) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx]: I feel like there has to be a more elegant way of navigating the user to the desired page, but for a first pass draft PR this felt like an okay starting point. Advice is very welcome on how to clean it up.

Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx]: We can use the `getSiteWooCommerceUrl` function.

```
import getSiteWooCommerceUrl from 'calypso/state/sites/selectors/get-site-woocommerce-url';

...

const siteAdminUrl = useSelector( ( state ) => getSiteWooCommerceUrl( state, siteId ) );
```

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx]: I'd like to suggest a few UX improvements here:
* Bold the email address so that it's more prominent.
* Apply a "busy" state to the button when the email is being sent (this ensures we remain consistent with buttons in other Congrats pages. See: https://github.com/Automattic/wp-calypso/blob/02c2989e213b75fc9bd24bc6db579ff720ea489d/client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-theme-section.tsx#L122) Note that we'll need to adjust the import from `@wordpress/components` to `@automattic/components` for the `<Button>` component in order take advantage of the visual treatment for the busy state.
* Make "click here" a link and call `resendEmail` when clicked. It seemed weird that we're showing the text "click here" when it isn't clickable.

https://github.com/Automattic/wp-calypso/assets/104910361/93c78c95-8224-44a6-8f41-cd28203ee92b

```suggestion
			subtitle = translate(
				"{{paragraph}}With the plan sorted, verify your email address to create your store.{{br/}}Please click the link in the email we sent to {{strong}}%(emailAddress)s{{/strong}}.{{/paragraph}}{{paragraph}}If you haven't received the verification email, please {{a}}click here{{/a}}.{{/paragraph}}",
				{
					args: { emailAddress: emailAddress },
					components: {
					    paragraph: <p />,
					    br: <br />,
					    strong: <strong />,
					    a: <a href="" onClick={ resendEmail },
					},
				}
			);
			const isSendingEmail = resendStatus === RESEND_PENDING;
			headerButtons = (
				<Button variant="secondary" onClick={ resendEmail } busy={ isSendingEmail } disabled={ isSendingEmail }>
					{ resendButtonText() }
				</Button>
			);
```

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx]: We typically declare constants/variables such that they're closer to the block of code that uses them, so it's obvious they are used for.

**Review by wongasy — CHANGES_REQUESTED** [review]
Looking good! Just a few comments.

**Comment by wongasy** [comment]
@chad1008 A heads up that I added the `String freeze` label to get the strings in for translation.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx]: This is excellent, thank you.

Applying the diff didn't quite work, as the linter threw an a11y warning on the anchor. Once it has an `onClick` it points out that we're using an anchor for a button when those are intended for navigation. I've set an ignore on the line for now. Would you prefer we ignore that lint rule in this case? Or use an actual `Button` with a combination of props and CSS to make it look right?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11571312

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @chad1008 for including a screenshot in the description! This is really helpful for our translators.

**Comment by chad1008** [comment]
Thanks for the tweaks @wongasy.  https://github.com/Automattic/wp-calypso/pull/88256 is merged onto this branch, and the CI checks are running now.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx]:
```suggestion
	if ( isWpComEcommercePlan( primaryPurchase.productSlug ) ) {
```
I missed this in the last round of review, we should be using the helper function to check for eCommerce plans rather than a hardcoded string. That ensures all types of eCommerce plans are caught regardless of payment interval. The `isWpComEcommercePlan` function is part of the `@automattic/calypso-products` package.

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/products/plan-product.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/redesign-v2/products/plan-product.tsx]:
```suggestion
					{ ! isWpComEcommercePlan( purchase.productSlug ) && (
```

Similarly here.

**Review by wongasy — APPROVED** [review]
LGTM and tests well! If we could address the last feedback regarding plan slug check, this should be good to go! Let's see where translation stands tomorrow, I'm inclined to ship this even when translation is not fully completed, given the re-send email functionality has been missing since plan-only redesign page went live 😞 

* Header text and button updated for eCommerce plan

Screen Shot 2024-03-07 at 1 40 58 PM

* Success notice shown when email resent successfully

https://github.com/Automattic/wp-calypso/assets/104910361/6c819a45-1844-4fc0-90f4-e855ab6acd92

* Error notice shown when email failed to be resent

https://github.com/Automattic/wp-calypso/assets/104910361/b62298ac-f338-44fc-b503-fd074ea2af09

File: client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx
Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/plan-only.tsx]: Oh, cool. Thanks for this... definitely helpful for me to see these as I familiarize myself with the different types of helper functions we have in place!

**Comment by chad1008** [comment]
Latest feedback re: the helper function is in. Waiting for CI!

**Comment by chad1008** [comment]
CI checks are green!

@wongasy, I'll hold off until tomorrow for translations as you suggested, but check in with you if they aren't ready to finalize whether or not we want to merge proactively to get the resend functionality back into the flow.

**Comment by wongasy** [comment]
@chad1008 It just occurred to me that we should also account for the logic here 😞 https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.tsx#L644-L666

I'm going to look at this tomorrow morning.

**Comment by wongasy** [comment]
>It just occurred to me that we should also account for the logic here 😞 https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.tsx#L644-L666
>
>I'm going to look at this tomorrow morning.

@chad1008 The main thing we're missing is the "loading" bar that shows the steps like "setting up your site...", etc. That bar is also missing in production, so we're not introducing new bugs here. The other piece that I haven't been able to figure out is `<WpAdminAutoLogin>` (https://github.com/Automattic/wp-calypso/blob/1a9c00a391a5d9c077f44bc8687de685c6e7576c/client/my-sites/checkout/checkout-thank-you/index.tsx#L658-L661), it isn't exactly clear to me as to what that does. That said, we never get to those lines of code in production anyways for single plan purchases, so again, we're not introducing bugs with our changes here.

Let's proceed to roll out and we can iterate on those missing pieces in followup PRs if needed.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11571312).

**Pull request merged**
